scripts/zones/RoMaeve/Zone.lua

```diff
@@ -35,21 +35,39 @@ function onRegionEnter(player, region)
 end
 
 function onGameHour(zone)
-    local hour = VanadielHour()
+    local vanadielHour = VanadielHour()
+    
+    -- Make Ro'Maeve come to life between 6pm and 6am during a full moon
+    if IsMoonFull() and (vanadielHour >= 18 or vanadielHour < 6) then
+        local moongate1 = GetNPCByID(ID.npc.MOONGATE_OFFSET)
+        local moongate2 = GetNPCByID(ID.npc.MOONGATE_OFFSET + 1)
 
-    if IsMoonFull() and hour == 18 then
-        for i = ID.npc.MOONGATE_OFFSET, ID.npc.MOONGATE_OFFSET+7 do
-            GetNPCByID(i):openDoor(1728)
-            GetNPCByID(i):untargetable(true)
+        if moongate1:getLocalVar("romaeveActive") == 0 then
```

Comment:
note to self: look into making variables on the zone object itself a thing. if for some reason this one NPC gets lost by something as trivial as an ID shift, the whole thing breaks.